grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -625,20 +679,12 @@ class StockMovementService {
 
     EditPage getEditPage(String id) {
         EditPage editPage = new EditPage()
-        def startTime = System.currentTimeMillis()
-        StockMovement stockMovement = getStockMovement(id)
-        log.info("Get stock movement ${id}: ${System.currentTimeMillis() - startTime}")
-
-        Map monthlyStocklistQuantities = calculateMonthlyStockListQuantity(stockMovement.origin)
 
-        startTime = System.currentTimeMillis()
-        stockMovement.lineItems.each { stockMovementItem ->
-            stockMovementItem.stockMovement = stockMovement
-            EditPageItem editPageItem = buildEditPageItem(stockMovementItem)
-            editPageItem.quantityConsumed = monthlyStocklistQuantities.get(stockMovementItem.product.id)
-            editPage.editPageItems.addAll(editPageItem)
+        if (!grailsApplication.config.openboxes.api.pagination.enabled) {
+            StockMovement stockMovement = getStockMovement(id)
+            editPage.editPageItems = getEditPageItems(stockMovement.lineItems, stockMovement.origin)
```

Comment:
I like that you moved this logic into a method, but I think an even better approach would be to pass the stock movement to the EditPage constructor and move the getEditPageItems() method into the EditPage class.
```
StockMovement stockMovement = getStockMovement(id)
return new EditPage(stockMovement)
```
Moving getEditPageItems() into EditPage would turn the editPageItems from "property" to "method", but it would work without changing the caller since editPage.editPageItems is equivalent to editPage.getEditPageItems() within Grails.

Reply:
Same applies for the other two pages (PickPage, PackPage).